components/ide/code/leeway.Dockerfile

```diff
@@ -19,17 +22,26 @@ RUN mkdir -p /ide/node/bin \
 RUN cp /ide/node/bin/node /ide/node/bin/gitpod-node && rm /ide/node/bin/node
 
 
-FROM mcr.microsoft.com/vscode/devcontainers/typescript-node:0-12 as code_installer
-
+FROM ubuntu:18.04 as code_installer
 
-# see https://github.com/gitpod-io/vscode/blob/bdeca3f8709b70b339f41fc2a14e94f83d6475ac/.github/workflows/ci.yml#L130
-RUN sudo apt-get update \
-    && sudo apt-get -y install --no-install-recommends libxkbfile-dev pkg-config libsecret-1-dev libxss1 dbus xvfb libgtk-3-0 libgbm1 \
+RUN apt-get update \
+    # see https://github.com/microsoft/vscode/blob/42e271dd2e7c8f320f991034b62d4c703afb3e28/.github/workflows/ci.yml#L94
+    && apt-get -y install --no-install-recommends libxkbfile-dev pkg-config libsecret-1-dev libxss1 dbus xvfb libgtk-3-0 libgbm1 \
+    && apt-get -y install --no-install-recommends git curl build-essential libssl-dev ca-certificates python \
     # Clean up
-    && sudo apt-get autoremove -y \
-    && sudo apt-get clean -y \
+    && apt-get autoremove -y \
+    && apt-get clean -y \
     && rm -rf /var/lib/apt/lists/*
 
+ARG NODE_VERSION
+ENV NVM_DIR /root/.nvm
+RUN curl -fsSL https://raw.githubusercontent.com/nvm-sh/nvm/v0.38.0/install.sh | sh \
+    && . $NVM_DIR/nvm.sh  \
+    && nvm install $NODE_VERSION \
+    && nvm alias default $NODE_VERSION \
+    && npm install -g yarn node-gyp
+ENV PATH $NVM_DIR/versions/node/v$NODE_VERSION/bin:$PATH
```

Comment:
Question: Why re-install Node.js via NVM here (with `nvm.sh` being quite slow) instead of doing something like this?

```Dockerfile
COPY --from=node_installer /ide/node /ide/node
ENV PATH="/ide/node/bin:$PATH"
```

This is unimportant though, as this NVM/Node.js install doesn't seem to be captured in the final `FROM scratch` image below.

As long as this works it's good as is. 👍

Reply:
Just did not think about it